pandas/io/parsers.py

```diff
@@ -2836,7 +2836,8 @@ def _rows_to_cols(self, content):
             for row_num, actual_len in bad_lines:
                 msg = ('Expected %d fields in line %d, saw %d' %
                        (col_len, row_num + 1, actual_len))
-                if len(self.delimiter) > 1 and self.quoting != csv.QUOTE_NONE:
+                if self.delimiter and \
+                   len(self.delimiter) > 1 and self.quoting != csv.QUOTE_NONE:
```

Comment:
Generally practice for is to use parentheses around the conditional and not to use the slash i.e.:

~~~python
if (self.delimiter and
  len(self.delimiter) > 1...)
~~~